backend/lcfs/web/api/allocation_agreement/repo.py

```diff
@@ -317,3 +323,80 @@ async def get_latest_allocation_agreement_by_group_uuid(
 
         result = await self.db.execute(query)
         return result.unique().scalars().first()
+
+    async def delete_all_for_report(
```

Comment:
issue: It looks like this is only inserting DELETE records, but it needs to handle the case where we are in the same version it should hard delete the rows.